### PR TITLE
 Adding record reader config/context param to record transformer

### PR DESCRIPTION
**Pull request by swaminathanmanish**
**Whats in the PR:**
Adding Record reader config/context to transform api, for transformer to use it during transformation of record.
**Why its needed:**
Custom transformers need to be perform transformations based on record reader configs (eg: file path). This change is to enable such transformations.

### CODE REVIEW

**Comment by swaminathanmanish** [comment]
@snleee , @klsince - Based on our discussion, please take a look.

**Review by snleee — APPROVED** [review]
LGTM

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/RecordTransformer.java [hunk 1]
```diff
@@ -43,4 +44,15 @@ default boolean isNoOp() {
...
   @Nullable
   GenericRow transform(GenericRow record);
+
+  /**
+   * Transforms a record based on some custom rules using record reader context.
+   * @param record Record to transform
+   * @return Transformed record, or {@code null} if the record does not follow certain rules.
+   */
+  @Nullable
+  default GenericRow transformUsingRecordReaderContext(GenericRow record,
```
Comment (klsince) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/RecordTransformer.java:54]: just call it `transform(record, config)`, as it's easy to tell them apart by param list.

I'd suggest to pass in RecordReaderConfig (the interface), and make RecordReaderFileConfig implement RecordReaderConfig (which is an empty interface right now)

Comment (swaminathanmanish) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/RecordTransformer.java]: Yes this makes sense to me.

**Review by klsince — APPROVED** [review]

**Pull request merged**